src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -305,125 +312,122 @@ private static void BindProperty(PropertyInfo property, object instance, IConfig
-                            if (!bindingPoint.IsReadOnly && newValue != null)
-                            {
-                                bindingPoint.SetValue(newValue);
-                            }
-                        }
-
-                        return;
-                    }
+                    // for getter-only collection properties that we can't add to, nothing more we can do
+                    return;
+                }
 
-                    // -----------------------------------------------------------------------------------------------------------------------------
-                    //                         |  bindingPoint |  bindingPoint |
-                    //       Interface         |     Value     |   IsReadOnly  |  Behavior
-                    // -----------------------------------------------------------------------------------------------------------------------------
-                    //  IDictionary<T>         |   not null    |  true/false   | Use the Value instance to populate the configuration
-                    //  IDictionary<T>         |     null      |     false     | Create Dictionary<T> instance to populate the configuration
-                    //  IDictionary<T>         |     null      |     true      | nothing
-                    //  IReadOnlyDictionary<T> | null/not null |     false     | Create Dictionary<K,V> instance, copy over existing values, and populate the configuration
-                    //  IReadOnlyDictionary<T> | null/not null |     true      | nothing
-                    // -----------------------------------------------------------------------------------------------------------------------------
-                    if (TypeIsADictionaryInterface(type))
+                // -----------------------------------------------------------------------------------------------------------------------------
+                //                  |  bindingPoint |  bindingPoint |
+                //     Interface    |     Value     |   IsReadOnly  |  Behavior
+                // -----------------------------------------------------------------------------------------------------------------------------
+                //  ISet<T>         |   not null    |  true/false   | Use the Value instance to populate the configuration
+                //  ISet<T>         |     null      |     false     | Create HashSet<T> instance to populate the configuration
+                //  ISet<T>         |     null      |     true      | nothing
+                //  IReadOnlySet<T> | null/not null |     false     | Create HashSet<T> instance, copy over existing values, and populate the configuration
+                //  IReadOnlySet<T> | null/not null |     true      | nothing
+                // -----------------------------------------------------------------------------------------------------------------------------
```

Comment:
These comments with the table of scenarios is really helpful

Reply:
Yup!
Don't mean to take credit here... the diff to this file is purely indentation due to removing the unneeded `if (config != null)` check above.

Reply:
Yeah the source gen impl followed this closely.